### ISSUE TITLE
Timetable: multi-day session block shows up in the wrong place

### ISSUE THREAD

**Issue by ThiefMaster** [indico/indico#3073]
This was probably caused by 672e1b0e0d02e0302541d29e4dae71de935d8f64 which is supposed to show session blocks that contain a day transition (usually due to the user changing the display timezone).

However, the timetable JS apparently only takes the time into account so a block starting at 23:30 on the previous day shows up as starting at the end of the next day instead of the beginning.

To reproduce it:

- Set event timezone to Australia/Sydney
- Set event dates so the event has two days
- Add session block on first day, from 16:30-18:00
- Add contribution on second day, e.g. from 15:00-16:00
- Go to the display view and set the timezone to America/Los_Angeles

Now the session block (23:30-01:00) shows up on the second day after the contribution, even though it's at the beginning of the day.

**Comment by akira-okumura** [comment]
Due to the COVID-19 circumstances, a lot of conference participants from different time zones join online meetings now. But changing the Indico time zone splits a single session to two different wrong locations.

In my case a single session held during 16:00–18:00 (CEST) on May 13 is shown as below when the time zone is set to Europer/Berlin (GMT +2).

### Shown as a single session when Europer/Berlin (GMT +2) is chosen.
<img width="741" alt="Screen Shot 2020-05-14 at 17 02 25" src="https://user-images.githubusercontent.com/2981330/81909641-daea1c00-9605-11ea-8aef-e225aa2b40ac.png">

But when Asia/Tokyo (GMT +2) is chosen, the latter half of the session is moved to 0:00 of May 13, while the correct time is 0:00 of May ***14***.

### Shown as split sessions when Asia/Tokyo (GMT +7) is chosen (first half)
<img width="751" alt="Screen Shot 2020-05-14 at 17 05 41" src="https://user-images.githubusercontent.com/2981330/81910001-5ea40880-9606-11ea-9fb3-497eac080187.png">

### This one should be located in the May 14 tab (second half)
<img width="753" alt="Screen Shot 2020-05-14 at 17 05 33" src="https://user-images.githubusercontent.com/2981330/81910007-5fd53580-9606-11ea-9d6a-151e5dccd58b.png">

As I am not the admin of this Indico server, I do not know if the running Indico is up-to-date version. But it looks that this bug has not been fixed yet because the issue is still open.

Could the developers please fix this issue? Same problems will frequently happen.

**Comment by DirkHoffmann** [comment]
> As I am not the admin of this Indico server, I do not know if the running Indico is up-to-date version. But it looks that this bug has not been fixed yet because the issue is still open.

We are not on the latest version, but certainly one one which is newer than this issue (2017).
Anyway, @ThiefMaster, if you could confirm that it is not fixed (or from where it should be), we can plan for an upgrade (not before the end of the CTA meeting though, @akira-okumura) or a patch.

**Comment by ThiefMaster** [comment]
I don't think it's been fixed.

**Comment by DirkHoffmann** [comment]
FYI, our current version is 2.1.8. 

The bug has quite simple symptoms (meeting is seen one day earlier), but is still annoying. Do you have plans to fix it in the team, or would you prefer external help, @ThiefMaster?

**Comment by ThiefMaster** [comment]
I don't think this bug is a priority for us right now - it might be only looked into when rewriting the timetable in React and getting rid of all the legacy code (currently [planned](https://getindico.io/roadmap/) for 2021).

So if someone is really willing to dig into the legacy timetable code and fix this (without breaking anything for complex timetables - thinking about parallel sessions etc.), we'd of course greatly appreciate it!

**Comment by alexander-held** [comment]
I just also ran into this bug: a single session in an event taking place in CST shows up as two sessions when viewed in CET, on both the day it should show up and the following day (the session goes until midnight in CET). Is there any known workaround for this? Everything is fine in the "indico style" view, but I wonder how I have not noticed this happening at conferences in the past, so maybe there is a way to avoid this?

**Comment by ThiefMaster** [comment]
There's no workaround (besides waiting for Indico v3.5 around ~~the end of this / beginning of next year)~~ ~~2026~~ 2027-2028 :/

**Comment by dguest** [comment]
I just hit this bug too. Thanks for the consideration and of course I understand if it's not the highest priority, just wanted to mention that I found it.
